src/main/java/JobData.java

```diff
@@ -99,7 +100,41 @@ public static ArrayList<HashMap<String, String>> findByValue(String value) {
         loadData();
 
         // TODO - implement this method
-        return null;
+        ArrayList<HashMap<String, String>> jobs = new ArrayList<>();
+
+        for (HashMap<String, String> job : allJobs) {
+
+            // MAKE SEARCH METHODS CASE-INSENSITIVE
+            // Making the "job" hashmap lowercase
+            HashMap<String, String> jobLowerCase = new HashMap<>();
+            for (Map.Entry<String, String> row : job.entrySet()) {
```

Comment:
Nice work on making this search case insensitive as well! I recommend checking out `keySet()` and how you can make this logic a little shorter. This approach works well, though!